core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -914,6 +920,9 @@ class GroupCoordinator(val brokerId: Int,
       if (isPending) {
         debug(s"Pending member $memberId has been removed after session timeout expiration.")
         group.removePendingMember(memberId)
+        if (group.is(CompletingRebalance)) {
+          heartbeatPurgatory.checkAndComplete(GroupKey(group.groupId))
```

Comment:
I am not sure this is right. It is the rebalance that may be blocked by this pending member. I think we want to do the same thing we did above.
```
if (group.is(PreparingRebalance)) {
  joinPurgatory.checkAndComplete(GroupKey(group.groupId))
}
```
By the way, are there any tests we could write for some of these fixes?